staging/src/k8s.io/client-go/tools/cache/shared_informer_health.go

```diff
@@ -0,0 +1,214 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cache
+
+import (
+	"sync"
+	"sync/atomic"
+	"time"
+)
+
+func newDefaultListWatchHealth(errorReceiverCh <-chan int) *listWatchHealth {
+	return &listWatchHealth{
+		minTimeToFail:       time.Duration(5 * time.Second),
+		minConsistentErrors: 10,
+		consistentErrorsCh:  errorReceiverCh,
+	}
+}
+
+type listWatchHealth struct {
```

Comment:
i'm going to remove this implementation if `listWatchLockFreeHealth` is good enough.